src/adagrams.js

```diff
@@ -1,15 +1,116 @@
+// import { LETTER_POOL } from '../test/adagrams.test.js';
+
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterList = [];
+  for (const [letter, frequency] of Object.entries(LETTER_POOL)) {
+    for (let i = 0; i < frequency; i++) {
+      letterList.push(letter);
+    }
+  }
+
+  const lettersInHand = [];
+  
+  while (lettersInHand.length < 10) {
+    const index = Math.floor(Math.random() * letterList.length);
+    lettersInHand.push(letterList[index]);
+    letterList.splice(index, 1);
+  }
+  return lettersInHand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const lettersInHandMap = new Map();
+
+  // Populate the Map with letter counts
+  for (const letter of lettersInHand) {
+    if (lettersInHandMap.has(letter)) {
+      lettersInHandMap.set(letter, lettersInHandMap.get(letter) + 1);
+    } else {
+      lettersInHandMap.set(letter, 1);
+    }
+  }
+
+  // Check if the input can be formed with the letters in hand
+  for (const letter of input) {
+    if (!lettersInHandMap.has(letter) || lettersInHandMap.get(letter) === 0) {
+      return false;
+    } else {
+      lettersInHandMap.set(letter, lettersInHandMap.get(letter) - 1);
+    }
+  }
```

Comment:
Great use of a frequency map to allow us to validate the word letters efficiently.

We could also calculate a frequency map for the word itself, and then check whether that map is "contained" within the hand map. An additional helper function would still keep the code here very clear.